app/validators/blocks/list_collector_validator.py

```diff
@@ -90,25 +106,52 @@ def validate_not_for_supplementary_list(self):
             )
 
     def validate_other_list_collectors(self):
+        """
+        Checks other list collectors for:
+            - non-unique answer id in add block for any other same-named list collectors
+            - duplicate answer id in add, edit, or remove block for other different-named list collectors
+        """
         list_name = self.block["for_list"]
-        add_answer_ids = self.questionnaire_schema.get_all_answer_ids(
-            self.block["add_block"]["id"]
-        )
+        list_blocks_answer_ids = {
+            child_block: self.questionnaire_schema.get_all_answer_ids(
+                self.block[child_block]["id"]
+            )
+            for child_block in self.LIST_CHILD_BLOCKS_TO_VALIDATE
+        }
 
         other_list_collectors = self.questionnaire_schema.get_other_blocks(
-            self.block["id"], for_list=list_name, type="ListCollector"
+            self.block["id"], type="ListCollector"
         )
 
         for other_list_collector in other_list_collectors:
-            other_add_ids = self.questionnaire_schema.get_all_answer_ids(
-                other_list_collector["add_block"]["id"]
-            )
-            difference = add_answer_ids.symmetric_difference(other_add_ids)
-            if difference:
-                self.add_error(
-                    self.NON_UNIQUE_ANSWER_ID_FOR_LIST_COLLECTOR_ADD,
-                    list_name=list_name,
+            other_list_blocks_answer_ids = {
+                child_block: self.questionnaire_schema.get_all_answer_ids(
+                    other_list_collector[child_block]["id"]
                 )
+                for child_block in self.LIST_CHILD_BLOCKS_TO_VALIDATE
+            }
+
+            if list_name == other_list_collector["for_list"]:
+                if list_blocks_answer_ids["add_block"].symmetric_difference(
+                    other_list_blocks_answer_ids["add_block"]
+                ):
+                    self.add_error(
+                        self.DIFFERENT_LIST_COLLECTOR_ADD_BLOCKS_FOR_SAME_LIST,
+                        list_name=list_name,
```

Comment:
I quite like that you have block id and other list collector block id in the error added below, adding that to this one too could be helpful

Reply:
Good idea, thanks